lib/classes/patch/410alpha10a.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+/**
+ * This file is part of Phraseanet
+ *
+ * (c) 2005-2016 Alchemy
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+use Alchemy\Phrasea\Application;
+use Alchemy\Phrasea\Core\Configuration\PropertyAccess;
+use Alchemy\Phrasea\Model\Manipulator\ApiApplicationManipulator;
+use Alchemy\Phrasea\Model\Entities\ApiApplication;
+
+
+class patch_410alpha10a implements patchInterface
+{
+    /** @var string */
+    private $release = '4.1.0-alpha.10';
+
+    /** @var array */
+    private $concern = [base::APPLICATION_BOX];
+
+    /**
+     * {@inheritdoc}
+     */
+    public function get_release()
+    {
+        return $this->release;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDoctrineMigrations()
+    {
+        return [];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function require_all_upgrades()
+    {
+        return false;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function concern()
+    {
+        return $this->concern;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function apply(base $databox, Application $app)
+    {
+        // create an api application for adobeCC
+
+        /** @var ApiApplicationManipulator $manipulator */
+        $manipulator = $app['manipulator.api-application'];
+
+        if($manipulator->findByClientID(\API_OAuth2_Application_adobeCCPlugin::CLIENT_ID) === null) {
+
```

Comment:
better to use
$app['repo.api-applications']->findByClientId(), so no need to add the method in manipulator